README
NAME
    Dancer::Serializer::MessagePack - MessagePack serializer for Dancer


VERSION
    version 0.01

SYNOPSIS
        # in your Dancer app:
        set serializer => 'MessagePack';

        # or in your Dancer config file:
        serializer: 'MessagePack'

DESCRIPTION
    This serializer serializes your data structure to MessagePack format. 
    
    It uses Data::MessagePack's "pack" and "unpack" functions.

SUBROUTINES/METHODS
  init
    An initializer that is called automatically by Dancer.

    Runs "loaded".

  loaded
    Lazily loads Data::MessagePack and imports the appropriate functions.

  serialize
    Serializes a given data to MessagePack format

  deserialize
    Deserializes a given data from MessagePack format.

  from_msgpack
    Helper function to create a new Dancer::Serializer::MessagePack object and
    run "serialize".

  to_msgpack
    Helper function to create a new Dancer::Serializer::MessagePack object and
    run "deserialize".

  content_type
    Returns the content type which is text/x-msgpack.

SEE ALSO
    Data::MessagePack
    https://github.com/msgpack/msgpack-perl

AUTHOR
      Nikhil Mulley <mnikhil@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2013 by Nikhil Mulley.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
0.01    08.21.2013
        MessagePack serializer plugin for Dancer application


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  'build_requires' => {
    'English' => '0',
    'File::Find' => '0',
    'File::Temp' => '0',
    'Module::Build' => '0.3601',
    'Test::Fatal' => '0',
    'Test::More' => '0'
  },
  'configure_requires' => {
    'ExtUtils::MakeMaker' => '6.31',
    'Module::Build' => '0.3601'
  },
  'dist_abstract' => 'MessagePack serializer for Dancer',
  'dist_author' => [
    'Nikhil Mulley <mnikhil@cpan.org>'
  ],
  'dist_name' => 'Dancer-Serializer-MessagePack',
  'dist_version' => '0.01',
  'license' => 'perl',
  'module_name' => 'Dancer::Serializer::MessagePack',
  'recommends' => {},
  'recursive_test_files' => 1,
  'requires' => {
    'Carp' => '0',
    'Dancer' => '0',
    'Dancer::Serializer::Abstract' => '0',
    'Data::MessagePack' => '0.46'
  },
  'script_files' => []
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


lib/Dancer/Serializer/MessagePack.pm
package Dancer::Serializer::MessagePack;

use strict;
use warnings;
use Carp;
use Dancer::ModuleLoader;
use Dancer::Deprecation;
use Dancer::Config 'setting';
use Dancer::Exception qw(:all);
use base 'Dancer::Serializer::Abstract';


# helpers

sub to_msgpack {
  my $s = Dancer::Serializer::MessagePack->new;
  return $s->serialize(@_);
}

sub from_msgpack {
  my $s = Dancer::Serializer::MessagePack->new;
  return $s->deserialize(@_);
}

# class definition

sub loaded { Dancer::ModuleLoader->load_with_params('Data::MessagePack') }

sub init {
  my ($self) = @_;
  raise core_serializer => 'Data::MessagePack is needed and is not installed'
    unless $self->loaded;
}

sub serialize {
  my $self   = shift;
  my $entity = shift;

  my $depth = 1024;
  return Data::MessagePack->pack($entity) || 
      croak "Unable to serialize the entity ! \n";
}

sub deserialize {
  my $self   = shift;
  my $entity = shift;

  return Data::MessagePack->unpack($entity) ||
      croak "Unable to deserialize the entity ! \n";
}

sub content_type {'application/x-msgpack'}

1;


#################### pod generated by Pod::Autopod - keep this line to make pod updates possible ####################

=head1 NAME

Dancer::Serializer::MessagePack


=head1 DESCRIPTION

    This serializer serializes your data structure to MessagePack format.

    It uses Data::MessagePack "pack" and "unpack" functions.

=head1 IMPLEMENTS

L<Dancer::Serializer::Abstract> 

=head1 SYNOPSIS

    # in your Dancer app:
    setting serializer => 'MessagePack';

    # or in your Dancer config file:
    serializer: 'MessagePack'

=head1 METHODS

=head2 init

    An initializer that is called automatically by Dancer.

    Runs "loaded".

=head2 loaded

    Lazily loads Data::MessagePack and imports the appropriate functions.

=head2 serialize

    Serializes a given data to MessagePack format

=head2 to_msgpack

    Helper function to create a new Dancer::Serializer::MessagePack object and
    run "serialize".

=head2 deserialize

    Deserializes a given data from MessagePack format.

=head2 from_msgpack

    Helper function to create a new Dancer::Serializer::MessagePack object and
    run "deserialize".

=head2 content_type

    Returns the content type which is application/x-msgpack.

=head1 SEE ALSO

    Data::MessagePack
    https://github.com/msgpack/msgpack-perl

=head1 AUTHOR

    Nikhil Mulley <mnikhil@cpan.org>

=head1 COPYRIGHT AND LICENSE

    This software is copyright (c) 2013 by Nikhil Mulley.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.

=cut



Makefile.PL
use strict;
use warnings;

use ExtUtils::MakeMaker 6.31;



my %WriteMakefileArgs = (
  'ABSTRACT' => 'MessagePack serializer for Dancer',
  'AUTHOR' => 'Nikhil Mulley <mnikhil@cpan.org>',
  'BUILD_REQUIRES' => {
    'English' => '0',
    'File::Find' => '0',
    'File::Temp' => '0',
    'Module::Build' => '0.3601',
    'Test::Fatal' => '0',
    'Test::More' => '0'
  },
  'CONFIGURE_REQUIRES' => {
    'ExtUtils::MakeMaker' => '6.31',
    'Module::Build' => '0.3601'
  },
  'DISTNAME' => 'Dancer-Serializer-MessagePack',
  'EXE_FILES' => [],
  'LICENSE' => 'perl',
  'NAME' => 'Dancer::Serializer::MessagePack',
  'PREREQ_PM' => {
    'Carp' => '0',
    'Dancer' => '0',
    'Dancer::Serializer::Abstract' => '0',
    'Data::MessagePack' => '0.46',
  },
  'VERSION' => '0.01',
  'test' => {
    'TESTS' => 't/*.t'
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



